src/generator/operations.ts

```diff
@@ -105,11 +105,15 @@ export async function generateOperations(session: Session<CodeModel>): Promise<O
 }
 
 // use this to generate the code that will help process values returned in response headers
-function formatHeaderResponseValue(propName: string, header: string, schema: Schema, imports: ImportManager, respObj: string, zeroResp: string): string {
+function formatHeaderResponseValue(propName: string, header: string, schema: Schema, imports: ImportManager, respObj: string, zeroResp: string, multPrefix: boolean): [string, boolean] {
   // dictionaries are handled slightly different so we do that first
   if (schema.type === SchemaType.Dictionary) {
     imports.add('strings');
-    let text = `\tprefix := strings.ToUpper("${schema.language.go!.headerCollectionPrefix}")\n`;
+    let firstPrefix = ':';
+    if (multPrefix === true) {
```

Comment:
nit: should this be `multiPrefix`?

Reply:
NIT:
```suggestion
    if (multPrefix) {
```

Reply:
Also, perhaps it is out of scope for this PR but not sure why the function has that many parameters, I would expect related ones to be grouped in objects instead.

Reply:
Done! Updated to multiPrefix and fixed the if. I will bring up your comment about the function signature once Joel is back so that we can discuss that. We havent really used the model of grouping params into objects evenly throughout the whole generator.